store/src/java/com/zimbra/cs/fb/ExchangeEWSFreeBusyProvider.java

```diff
@@ -920,42 +923,126 @@ public boolean handleMailboxChange(String accountId) {
             String fb = "";
             int i = 0;
             for (FreeBusyResponseType attendeeAvailability : results) {
-                if (re.email == attendees.getMailboxData().get(i).getEmail()
-                        .getAddress()) {
-                    if (ResponseClassType.SUCCESS != attendeeAvailability
-                            .getResponseMessage().getResponseClass()) {
-                        ZimbraLog.fb
-                                .warn(attendeeAvailability.getResponseMessage().getMessageText());
-                        ZimbraLog.fb
-                                .warn("Error in response. continuing to next one");
-
-                        i++;
-                        continue;
-                    }
-                    ZimbraLog.fb.debug("Availability for "
-                            + attendees.getMailboxData().get(i).getEmail()
-                                    .getAddress()
-                            + " ["
-                            + attendeeAvailability.getFreeBusyView()
-                                    .getMergedFreeBusy() + "]");
-
-                    fb = attendeeAvailability.getFreeBusyView().getMergedFreeBusy();
-                    if (fb == null) {
-                        ZimbraLog.fb.warn("Merged view Free Busy info not avaiable");
-                        fb = "";  //Avoid NPE.
+            	if (attendeeAvailability.getFreeBusyView() != null) {
+                	String fbResponseViewType = attendeeAvailability.getFreeBusyView().getFreeBusyViewType().get(0);
+                	String emailAddress = attendees.getMailboxData().get(i).getEmail().getAddress();
+                	String name = attendees.getMailboxData().get(i).getEmail().getName();
+                	ZimbraLog.fb.debug("Exchange EWS free busy view type from response : %s", fbResponseViewType);
+                    if (re.email == emailAddress) {
```

Comment:
better to use equals()
```suggestion
                    if (re.email.equals(emailAddress)) {
```